packages/mongo/tests/observe_changes_tests.js

```diff
@@ -519,8 +519,8 @@ if (Meteor.isServer) {
       const [resolver1, promise1] = getPromiseAndResolver();
       const [resolver2, promise2] = getPromiseAndResolver();
 
-      await self.insert({x: 2, y: 3});
       self.expects.push(resolver1, resolver2);
+      await self.insert({x: 2, y: 3});
```

Comment:
Does the order matter here? I think it does not, but I'm interested _why_ this change was made.

Reply:
The order matter because if you call `insert` before add the resolver inside the  `self.expects` array, the `added` callback breaks since `self.expects` is empty` 
<img width="619" height="165" alt="Screenshot 2025-11-10 at 11 36 10 AM" src="https://github.com/user-attachments/assets/f6a07977-5ff7-40a5-a0a1-3df4065f6134" />

maybe it doent matter for oplog due a latency issue

Reply:
Then I'd extract this change to the pre-merged tests, too. Just to make sure it passed before as well.

Reply:
this fix comes with change streams context, since it doesnt matter for oplog it should be included in this PR IMO, to mantain the context, would be useful to know this change comes after change streams